Tests/Get-BMDeployment.Tests.ps1

```diff
@@ -212,9 +211,177 @@ Describe 'Get-BMDeployment' {
+    }
+
+    It 'should get deployment by release id' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Release $script:releaseRelease.id
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 1
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldNotBeReturned $deployment2
+        ThenDeploymentShouldBeReturned $deployment3
+    }
+
+    It 'should get deployment by release name' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Release $script:releaseRelease.name
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 1
+        ThenDeploymentShouldNotBeReturned $deployment
+        ThenDeploymentShouldNotBeReturned $deployment2
+        ThenDeploymentShouldBeReturned $deployment3
+    }
+
+    It 'should get deployment by build object' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Build $build
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 2
+        ThenDeploymentShouldBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+    }
+
+    It 'should get deployment by build id' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Build $build.id
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 2
+        ThenDeploymentShouldBeReturned $deployment
+        ThenDeploymentShouldBeReturned $deployment2
+        ThenDeploymentShouldNotBeReturned $deployment3
+    }
+
+    It 'should get deployment by pipeline name' {
+        $build = GivenReleaseBuild $script:releaseAll
+        $build2 = GivenReleaseBuild $script:releaseRelease
+        $deployment = GivenDeployment $build
+        $deployment2 = GivenDeployment $build -Stage 'Testing'
+        $deployment3 = GivenDeployment $build2
+        WhenGettingBMDeployment -Pipeline $script:pipeline.Pipeline_Name
+        ThenShouldNotThrowErrors
+        ThenTotalDeploymentsReturned 2

```

Comment:
I don't see how this test sets up deployments of different pipelines? In its current state, the test should assert 3 deployments returned, not 2.